src/components/tl/ucp/tl_ucp_context.c

```diff
@@ -13,6 +13,99 @@
 #include "schedule/ucc_schedule_pipelined.h"
 #include <limits.h>
 
+static inline ucc_status_t ucc_tl_ucp_context_service_init(
```

Comment:
i would try to unify this function for both workers to reduce code duplication

Reply:
Sure. I did what I could in the separate commit `TL/UCP: refactor context init`. Lmk if you think this could be done better